### YamlMime:ManagedReference
items:
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0d423877466065f42d44d66a2d881615
    id: _provis0d423877466065f42d44d66a2d881615
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    children:
      - 'com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0d423877466065f42d44d66a2d881615.httpResponseVerification(int,String)'
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0d423877466065f42d44d66a2d881615.yml
    langs:
      - java
    name: ProvisioningServiceClientExceptionManager
    nameWithType: ProvisioningServiceClientExceptionManager
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientExceptionManager
    type: Class
    source:
      remote: &ref_0
        path: provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/exceptions/ProvisioningServiceClientExceptionManager.java
        branch: master
        repo: 'https://github.com/Azure/azure-iot-sdk-java'
      path: provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/exceptions/ProvisioningServiceClientExceptionManager.java
      startLine: 12
    package: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    summary: <p>Geben Sie die statische Funktion zum Überprüfen der Ergebnisse der Registrierungsdienst für Bereitstellung und die entsprechende Ausnahme ausgelöst. </p>
    syntax:
      content: public class ProvisioningServiceClientExceptionManager
    inheritance:
      - java.lang.Object
  - uid: 'com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0d423877466065f42d44d66a2d881615.httpResponseVerification(int,String)'
    id: 'httpResponseVerification(int,String)'
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0d423877466065f42d44d66a2d881615
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0d423877466065f42d44d66a2d881615.yml
    langs:
      - java
    name: 'httpResponseVerification(int responseStatus, String errorReason)'
    nameWithType: 'ProvisioningServiceClientExceptionManager.httpResponseVerification(int responseStatus, String errorReason)'
    fullName: 'static void com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientExceptionManager.httpResponseVerification(int responseStatus, String errorReason)'
    overload: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0d423877466065f42d44d66a2d881615.httpResponseVerification*
    type: Method
    source:
      remote: *ref_0
      path: provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/exceptions/ProvisioningServiceClientExceptionManager.java
      startLine: 47
    package: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    summary: |-
      <p>Überprüfen Sie die Antwort unter Verwendung des Response-status</p>
      <p>

        <pre>

          <code>ProvisioningServiceClientServiceException [any exception reported in the http response]

              |

              |

              +--&gt;ProvisioningServiceClientBadUsageException [any http response 4xx]

              |        |

              |        +--&gt;ProvisioningServiceClientBadFormatException [400]

              |        +--&gt;ProvisioningServiceClientUnathorizedException [401]

              |        +--&gt;ProvisioningServiceClientNotFoundException [404]

              |        +--&gt;ProvisioningServiceClientPreconditionFailedException [412]

              |        +--&gt;ProvisioningServiceClientTooManyRequestsException [429]

              |

              +--&gt;ProvisioningServiceClientTransientException [any http response 5xx]

              |        |

              |        +--&gt;ProvisioningServiceClientInternalServerErrorException [500]

              |

              +--&gt;ProvisioningServiceClientUnknownException [any other http response &gt;300, but not 4xx or 5xx]

          </code>

        </pre>

      </p>
      <p></p>
    syntax:
      content: 'public static void httpResponseVerification(int responseStatus, String errorReason)'
      parameters:
        - id: responseStatus
          type: f75371fa
          description: <p>ist der Status "Antwort" </p>
        - id: errorReason
          type: '26831127'
          description: <p>fehlerbeschreibung </p>
    exceptions:
      - type: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisec31e828069304f763e053102a181f40
        description: '<p>Diese Ausnahme wird ausgelöst, wenn der Antwortstatus 400 gleich </p>'
      - type: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis428e63329f7d8f9903aef432fc5a668d
        description: '<p>Diese Ausnahme wird ausgelöst, wenn der Antwortstatus 401 gleich </p>'
      - type: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis6cc9774bace2e6db41a3904df348eee0
        description: '<p>Diese Ausnahme wird ausgelöst, wenn der Antwortstatus 404 gleich </p>'
      - type: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf4a87e2786406409f327606adae377d
        description: '<p>Diese Ausnahme wird ausgelöst, wenn der Antwortstatus 412 gleich </p>'
      - type: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8a4878881f4ac3e4806ac2bc0f67ca18
        description: '<p>Diese Ausnahme wird ausgelöst, wenn der Antwortstatus 500 gleich </p>'
      - type: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9
        description: '<p>Diese Ausnahme wird ausgelöst, wenn der Antwortstatus keines diesen oben und dann größere 300 </p>'
references:
  - uid: f75371fa
    spec.java:
      - name: int
        fullName: int
  - uid: '26831127'
    spec.java:
      - name: String
        fullName: String
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis0d423877466065f42d44d66a2d881615.httpResponseVerification*
    name: httpResponseVerification
    nameWithType: ProvisioningServiceClientExceptionManager.httpResponseVerification
    fullName: static void com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientExceptionManager.httpResponseVerification
    package: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf54b54351d86c7bee42822d640658c9.yml
    name: ProvisioningServiceClientServiceException
    nameWithType: ProvisioningServiceClientServiceException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientServiceException
    type: Class
    summary: '<p>Dies ist die Teilmenge der Registrierungsdienst für Bereitstellung Ausnahmen für die Ausnahmen, die vom Dienst. </p>'
    syntax:
      content: public class ProvisioningServiceClientServiceException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisec31e828069304f763e053102a181f40
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisec31e828069304f763e053102a181f40.yml
    name: ProvisioningServiceClientBadFormatException
    nameWithType: ProvisioningServiceClientBadFormatException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientBadFormatException
    type: Class
    summary: |-
      <p>Erstellen von fehlerhaften Nachrichten-Format-Ausnahme</p>
      <p>Der Text der HTTP-Anforderung ist nicht gültig. Beispielsweise kann nicht analysiert werden, oder das Objekt kann nicht überprüft werden. </p>
      <p>HTTP-Statuscode 400. </p>
    syntax:
      content: public class ProvisioningServiceClientBadFormatException extends ProvisioningServiceClientBadUsageException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis428e63329f7d8f9903aef432fc5a668d
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis428e63329f7d8f9903aef432fc5a668d.yml
    name: ProvisioningServiceClientUnathorizedException
    nameWithType: ProvisioningServiceClientUnathorizedException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientUnathorizedException
    type: Class
    summary: |-
      <p>Nicht autorisierte Ausnahme erstellen</p>
      <p>Das Autorisierungstoken kann nicht überprüft werden; Angenommen, es ist abgelaufen oder gilt nicht für die Requestâ€™ s URI. Mit diesem Fehlercode wird auch als Teil des Flusses Nachweis TPM auf Geräten zurückgegeben. </p>
      <p>HTTP-Statuscode 401 </p>
    syntax:
      content: public class ProvisioningServiceClientUnathorizedException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis6cc9774bace2e6db41a3904df348eee0
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis6cc9774bace2e6db41a3904df348eee0.yml
    name: ProvisioningServiceClientNotFoundException
    nameWithType: ProvisioningServiceClientNotFoundException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientNotFoundException
    type: Class
    summary: |-
      <p>Erstellen von Device Bereitstellung Service, die mit der Ausnahme nicht gefunden</p>
      <p>Die Dienstinstanz Gerät bereitstellen oder eine Ressource (z. B. eine Registrierung) ist nicht vorhanden. </p>
      <p>HTTP-Statuscode 404. </p>
    syntax:
      content: public class ProvisioningServiceClientNotFoundException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf4a87e2786406409f327606adae377d
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provisdf4a87e2786406409f327606adae377d.yml
    name: ProvisioningServiceClientPreconditionFailedException
    nameWithType: ProvisioningServiceClientPreconditionFailedException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientPreconditionFailedException
    type: Class
    summary: |-
      <p>Erstellen Sie Vorbedingung nicht erfüllt-Ausnahme</p>
      <p>Das ETag in der Anforderung entspricht nicht das ETag der vorhandenen Ressource gemäß RFC7232. </p>
      <p>HTTP-Statuscode 412. </p>
    syntax:
      content: public class ProvisioningServiceClientPreconditionFailedException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8a4878881f4ac3e4806ac2bc0f67ca18
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8a4878881f4ac3e4806ac2bc0f67ca18.yml
    name: ProvisioningServiceClientInternalServerErrorException
    nameWithType: ProvisioningServiceClientInternalServerErrorException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientInternalServerErrorException
    type: Class
    summary: |-
      <p>Interne Serverausnahme für Fehler erstellen</p>
      <p>Ein interner Fehler ist aufgetreten. </p>
      <p>HTTP-Statuscode: 500. </p>
    syntax:
      content: public class ProvisioningServiceClientInternalServerErrorException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8d640264fa5997294ef7a61f3c9f7e67
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8d640264fa5997294ef7a61f3c9f7e67.yml
    name: ProvisioningServiceClientBadUsageException
    nameWithType: ProvisioningServiceClientBadUsageException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientBadUsageException
    type: Class
    summary: |-
      <p>Dies ist die Teilmenge der Registrierungsdienst für Bereitstellung Ausnahmen für die Ausnahmen, die im Zusammenhang mit Anwendungsfehler.</p>
      <p>Bereitstellung Service wird Antwort eine HTTP-Anforderung mit einem der Ausnahme Ungültige Verwendung, wenn die Rest-API mit einem Problem beim Ressourceneinsatz ordnungsgemäß aufgerufen wurde, z. B. der Benutzer für diesen Vorgang nicht autorisiert ist. </p>
      <p>HTTP-Statuscode 400 bis 499. </p>
    syntax:
      content: public class ProvisioningServiceClientBadUsageException extends ProvisioningServiceClientServiceException
  - uid: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8a0fa33fb6d65f102e28e9bcde677ce1
    parent: com.microsoft.azure.sdk.iot.provisioning.service.exceptions
    href: com.microsoft.azure.sdk.iot.provisioning.service.exceptions._provis8a0fa33fb6d65f102e28e9bcde677ce1.yml
    name: ProvisioningServiceClientTransientException
    nameWithType: ProvisioningServiceClientTransientException
    fullName: com.microsoft.azure.sdk.iot.provisioning.service.exceptions.ProvisioningServiceClientTransientException
    type: Class
    summary: |-
      <p>Dies ist die Teilmenge der Registrierungsdienst für Bereitstellung Ausnahmen die Ausnahmen einen temporären Dienstproblem beziehen.</p>
      <p>Der Bereitstellung-Dienst wird Antwort eine HTTP-Anforderung mit einem der vorübergehenden Ausnahme, wenn die Rest-API wurde ordnungsgemäß aufgerufen, aber der Dienst ist nicht mit dieser Aktion zu diesem Zeitpunkt ausführen. Hierbei handelt es sich um die Ausnahmen, die ein erneuter Versuch zur Behebung des Problems helfen. </p>
      <p>HTTP-Statuscode: 500 zu 599. </p>
    syntax:
      content: public class ProvisioningServiceClientTransientException